tests/syscalls.rs

```diff
@@ -59,15 +59,23 @@ fn test_contract<'a>(
         )
     });
 
-    let contract_state = ContractState::new(
-        class_hash,
-        tx_execution_context.nonce().clone(),
-        Default::default(),
-    );
-    let mut state_reader = InMemoryStateReader::new(HashMap::new(), HashMap::new());
+    let nonce = Felt::zero();
```

Comment:
`let nonce = tx_execution_context.nonce().clone();`
+ omit the storage inserts

Reply:
Done
> let nonce = tx_execution_context.nonce().clone();
I'm curious, why is this necessary?

Reply:
Just to keep the state as it was originally set

```
let contract_state = ContractState::new(
        class_hash,
        tx_execution_context.nonce().clone(),
        Default::default(),
    );
```